### PR TITLE
tsify SelectionRangeMask, utils, RowsContainer, RowGroup

### PR DESCRIPTION
**Pull request by MayhemYDG**

### CODE REVIEW

**Comment by MayhemYDG** [comment]
I think we should just import Immutable instead of relying on `window.Immutable`.

File: packages/react-data-grid/src/RowGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import React, { forwardRef } from 'react';
+import { last } from './common/utils';
+import { EventTypes } from './common/constants';
+import { Column, CellMetaData } from './common/types';
+import EventBus from './masks/EventBus';
+
+interface Props {
+  height: number;
+  columns: Column[];
+  row: unknown;
```
Comment (MayhemYDG) [inline packages/react-data-grid/src/RowGroup.tsx]: We'll need to overhaul these types at some point.

File: packages/react-data-grid/src/common/utils/index.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+// eslint-disable-next-line
+declare var Immutable: any;
```
Comment (MayhemYDG) [inline packages/react-data-grid/src/common/utils/index.ts]: I think we should just import Immutable and not rely on `window.Immutable`.
This file might need an overhaul, don't think we really need this `isFunction`.

Comment (amanmahajan7) [reply packages/react-data-grid/src/common/utils/index.ts]: That would be better and we can add `ImmutableJS` as a peer dependency. I also wonder if we even need to support immutable js or not.

**Pull request merged**